sycl/doc/extensions/SRGB/srgb_support.asciidoc

```diff
@@ -0,0 +1,163 @@
+= srgb_support
```

Comment:
I understand that this is an extension of the old SYCL 1.2.1 image API (not the new image API in SYCL 2020), however, I think it should still follow the SYCL 2020 extension guidelines.  Therefore, please use the extension spec for [device-if](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/DeviceIf/device_if.asciidoc) as a template for this extension.  (The "Hypothetical implementation" section in that spec is optional, so it is not necessary to have a section like that.)

In particular:

* Decide if this is an "intel" vs. a "oneapi" extension.  Generally, extensions that are specific to Intel hardware are "intel" extensions and others are "oneapi" extensions.  Therefore, I presume this is a "oneapi" extension.

* Decide on a feature-test macro name for this extension of the form `SYCL_EXT_ONEAPI_FOO` where "FOO" is some name for this extension.

* Use the feature-test macro name as the title of the extension instead of the " name string".  Remove the "name string" section from the spec.

Since it seems like this extension is only supported on some devices, there shoiuld be some way for an application to query whether a device supports the feature.  We generally use aspects for this.  Therefore, add an aspect named something like `ext_oneapi_foo`.  Probably the "foo" string would be the same name you use for the feature-test macro (but in lower case).

Finally, the spec should say what happens if an application violates the various limitations.  I think this is as simple as saying that the image constructor throws a `sycl::exception` with the `errc::invalid` error code for an invalid combination of parameters.  I think some of the limitations can't be diagnosed until the application creates an accessor?  If so, you can throw `sycl::exception` from the function that creates the accessor.

Reply:
Ok. Update the asciidoc template and reworked, added a feature test macro as well as support for an aspect, testing and docs for both. Spec mentions the error you'll receive if you don't observe the limitations.